src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioWorkspaceStatusServiceFactory.cs

```diff
@@ -148,30 +148,19 @@ public async Task WaitUntilFullyLoadedAsync(CancellationToken cancellationToken)
                 }
             }
 
-            // unfortunately, IVsOperationProgressStatusService requires UI thread to let project system to proceed to next stages.
-            // this method should only be used with either await or JTF.Run, it should be never used with Task.Wait otherwise, it can
-            // deadlock
-            public async Task<bool> IsFullyLoadedAsync(CancellationToken cancellationToken)
+            public bool IsFullyLoaded
             {
-                var status = await GetProgressStageStatusAsync(cancellationToken).ConfigureAwait(false);
-                if (status == null)
+                get
                 {
-                    return false;
-                }
+                    if (!_progressStageStatus.IsCompleted)
+                        return false;
 
-                return !status.IsInProgress;
-            }
+                    var status = _progressStageStatus.Task.CompletedResult();
+                    if (status is null)
+                        return false;
 
-            private async ValueTask<IVsOperationProgressStageStatusForSolutionLoad?> GetProgressStageStatusAsync(CancellationToken cancellationToken)
-            {
-                // Workaround for lack of fast path in JoinAsync; avoid calling when already completed
-                // https://github.com/microsoft/vs-threading/pull/696
-                if (_progressStageStatus.Task.IsCompleted)
-                {
-                    return await _progressStageStatus.Task.ConfigureAwait(false);
+                    return status.IsInProgress;
```

Comment:
this reads super weird to me.  we're "fully loaded" if we are "in progress"?  can you doc this?

Reply:
➡️ Unintentional negation; now fixed

Reply:
oh ok.  that makes much more sense.  thanks :)